rotkehlchen/chain/evm/decoding/hop/decoder.py

```diff
@@ -107,6 +112,44 @@ def _generate_bridge_note(
 
         return f'Bridge {amount} {asset.symbol_or_name()} {chain_label}{target_str}via Hop protocol'  # noqa: E501
 
+    def _process_hop_lp_token(self, lp_token: EvmToken, pool_address: ChecksumEvmAddress) -> None:
+        """Save the protocol value of the LP token and cache its pool address."""
+        if lp_token.protocol is None:
+            GlobalDBHandler.edit_evm_token(
+                entry=EvmToken.initialize(
+                    address=lp_token.evm_address,
+                    chain_id=self.evm_inquirer.chain_id,
+                    token_kind=EvmTokenKind.ERC20,
+                    name=lp_token.name,
+                    symbol=lp_token.symbol,
+                    decimals=lp_token.decimals,
+                    protocol=HOP_PROTOCOL_LP,
+                ),
+            )
```

Comment:
I just saw that the logic here is very complicated and error prone. You are initializating the lp token and you might or might not be missing the underlying tokem, then calling trhe edit evm token that are 3 calls at least only to change a value and so on. I would just do a single sql query

```
UPDATE evm_tokens SET protocol=? WHERE identifier=?
```

with `(HOP_PROTOCOL_LP, lp_token.identifier)`. Maybe for the sake of not having logic split everywhere you could create a method that does that + the cache cleaning